src/metabase/channel/render/js/svg.clj

```diff
@@ -43,22 +44,33 @@
     (js.engine/load-resource bundle-path)
     (js.engine/load-resource interface-path)))
 
-(def ^:private static-viz-context-delay
-  "Delay containing a graal js context. It has the chart bundle and the above `src-api` in its environment suitable for
-  creating charts. However, under some circumstances, the Truffle JS engine tends to leak memory, so we don't want to
-  keep the reference to the engine forever. Ideally, we would like to make the engine lifetime bound to a single
-  request or something like that, but that would introduce a lot of changes and noise to the system. Hence, with this
-  bandaid, we keep the JS engine for some reasonable time (10 minutes) before recreating it."
-  (delay/delay-with-ttl (.toMillis TimeUnit/MINUTES 10)
-                        #(load-viz-bundle (js.engine/context))))
-
-(defn- context
-  "Returns a static viz context. In dev mode, this will be a new context each time. In prod or test modes, it will
-  return the derefed contents of `static-viz-context-delay`."
-  ^Context []
+(def ^:private ^Pool static-viz-context-pool
+  "Pool of Truffle JS engine objects. They are not thread-safe, so the access to them has to be carefully managed
+  between threads. Each engine with loaded static viz code takes ~130 MB in memory, so we don't want too many of them.
+  However, one takes ~3 seconds to initialize, so we don't want to load them anew each time in prod. Under some
+  circumstances, the Truffle JS engine tends to leak memory, so we don't want to keep the reference to the engine
+  forever. Considering all that, this pool targets 100% utilization (so, if the utilization is lower, the pool will
+  start dropping objects) and the maximum of 3 objects (to prevent OOMs)."
+  (Pool. (reify IPool$Generator
```

Comment:
should we wrap it with delay?

Reply:
It's not necessary, this pool by itself doesn't produce anything until triggered, so it is safe to define it in top level.